acceptance-tests/src/test/java/tech/pegasys/web3signer/dsl/utils/MetadataFileHelpers.java

```diff
@@ -179,6 +183,42 @@ public void createAzureKeyYamlFileAt(
     }
   }
 
+  public void createYubiHsmYamlFileAt(
+      final Path metadataFilePath,
+      final Path destinationYubiShellSimulator,
+      final KeyType keyType) {
+    final int opaqueObjId = keyType == KeyType.BLS ? 1 : 2;
+
+    final Map<String, String> yaml = new HashMap<>();
+    yaml.put("type", "yubihsm2");
+    // TODO: Remove bin from yaml - Use env var
+    yaml.put("yubiShellBinaryPath", destinationYubiShellSimulator.toString());
+    yaml.put("connectorUrl", "http://localhost:12345");
+    yaml.put("authKey", String.valueOf(1));
+    yaml.put("password", "password");
+    yaml.put("opaqueObjId", String.valueOf(opaqueObjId));
+    yaml.put("keyType", keyType.name());
+
+    createYamlFile(metadataFilePath, yaml);
+  }
+
+  public static Path copyYubiHsmSimulator(final Path destDir) throws IOException {
+    final Path sourceYubiShellSimulator =
+        Path.of(Resources.getResource("YubiShellSimulator").getPath());
+    final Path destinationYubiShellSimulator = destDir.resolve("YubiShellSimulator");
```

Comment:
Could we just run the YubiShellSimulator in a thread so we have to don't have mess with copying files etc.

Reply:
we can't. The Signer's YubiHSM class needs to invoke it as an external process. I can create a small native binary instead?